redis/src/cluster_client.rs

```diff
@@ -434,6 +446,21 @@ impl ClusterClient {
         cluster::ClusterConnection::new(self.cluster_params.clone(), self.initial_nodes.clone())
     }
 
+    /// Creates new connections to Redis Cluster nodes with a custom config and returns a
+    /// [`cluster_async::ClusterConnection`].
+    ///
+    /// # Errors
+    ///
+    /// An error is returned if there is a failure while creating connections or slots.
+    pub fn get_connection_with_config(
```

Comment:
if the config is for the creation of async connections, why add this?,

Reply:
These fields:
```rust
    pub(crate) connection_timeout: Option<Duration>,
    pub(crate) response_timeout: Option<Duration>,
```
can also be applied  to a sync connection AFAIK. So I was thinking that it would be nice to also have this to match the async version. However I won't use it myself so I can remove this part.

Reply:
keep it, but adjust the config's documentation to remove the async part.